microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -363,6 +375,66 @@ def __init__(
         self.testpmd = _testpmd
         self.node = _node
         self.nic_controller = _node.features[NetworkInterface]
+        self.dmesg = _node.tools[Dmesg]
+        self._last_dmesg = ""
+        test_nic = self.node_nic_info.get_test_nic()[1]
+        # generate hotplug pattern for this specific nic
+        self.vf_hotplug_regex = re.compile(
+            f"{test_nic.upper}: Data path switched to VF: {test_nic.lower}"
+        )
+        self.vf_slot_removal_regex = re.compile(f"VF unregistering: {test_nic.lower}")
+
+    def wait_for_dmesg_output(self, wait_for: str, timeout: int) -> bool:
+        search_pattern = None
+        if wait_for == "AN_DISABLE":
+            search_pattern = self.vf_slot_removal_regex
+        elif wait_for == "AN_REENABLE":
+            search_pattern = self.vf_hotplug_regex
+        else:
+            raise LisaException(
+                "Unknown search pattern specified in "
+                "DpdkTestResources:wait_for_dmesg_output"
+            )
+
+        self.node.log.info(search_pattern.pattern)
+        time_accumulator = 0.0
+        while time_accumulator < timeout:
+            start = time.perf_counter()
+            output = self.dmesg.get_output(force_run=True)
+            stop = time.perf_counter()
+            if search_pattern.search(output.replace(self._last_dmesg, "")):
+                self._last_dmesg = output  # save old output to filter next time
+                self.node.log.info(
+                    f"Found VF hotplug info after {time_accumulator} seconds"
+                )
+                return True
+            time_accumulator += stop - start
```

Comment:
There can be a interval sleep like 1 sec, it avoids to run the function too fast.